api4/cloud.go

```diff
@@ -409,7 +410,33 @@ func sendAdminUpgradeRequestEmail(c *Context, w http.ResponseWriter, r *http.Req
 		return
 	}
 
-	if err = c.App.SendAdminUpgradeRequestEmail(user.Username, sub); err != nil {
+	if err = c.App.SendAdminUpgradeRequestEmail(user.Username, sub, model.InviteLimitation); err != nil {
+		c.Err = model.NewAppError("Api4.sendAdminUpgradeRequestEmail", err.Id, nil, err.Error(), err.StatusCode)
+		return
+	}
+
+	ReturnStatusOK(w)
+}
+
+func sendAdminUpgradeRequestEmailOnJoin(c *Context, w http.ResponseWriter, r *http.Request) {
+	if c.App.Srv().License() == nil || !*c.App.Srv().License().Features.Cloud {
+		c.Err = model.NewAppError("Api4.sendAdminUpgradeRequestEmailOnJoin", "api.cloud.license_error", nil, "", http.StatusNotImplemented)
+		return
+	}
+
+	user, err := c.App.GetUser(c.App.Session().UserId)
```

Comment:
In this case that the user is trying to join, they currently don't have user session given they haven't created their account yet. The email template does not include the username

Reply:
Is there another way to get something like their ```id``` or ```username```. Need one of those as key to track the rate limiting

Reply:
Yeah, that's a good point @marianunez for some reason I thought the request was coming from inside the installation.

In this case, we're doing a request to the API without authentication so you need to change the type of handler from `APISessionRequired` to `APIHandler`. Also given that you're not authenticated and you haven't created any user you won't have the username or the id so is not possible to pass that information to the mail

Reply:
@ethervoid any suggestions on how we can handle the rate limiting then?

Reply:
@ethervoid we're going to limit to 1 email per day. So the first user to encounter a block to invitation or join will trigger the sending of email to admin. Other users after that will find the call to that endpoint rate limited. We're also going to be rate limting based on a combination of the date and action (join/invitation)
cc: @marianunez

Reply:
Ok, if we limit to 1 per day by action and date looks good to me :)